### PR TITLE
Remove `referrerPolicy` from Cloudflare Worker `fetch` requests

### PR DESCRIPTION
**Pull request by DellaBitta**
### Discussion

The Cloudflare Worker runner runtime doesn't support the `fetch` parameter `referrerPolicy` and throws if one is defined. In attempt to better support Cloudlfare, we will remove this parameter from `fetch` requests if we detect the SDK is executing withing a Cloudflare runner enviornment.

### Testing

CI. Updated unit tests to test if `referrerPolicy` is omitted if `isCloudflareWorker()` returns true.

### API Changes

N/A.

### CODE REVIEW

**Review by NhienLam — APPROVED** [review]

File: .changeset/khaki-numbers-nail.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@firebase/auth': minor
```
Comment (hsubox76) [inline .changeset/khaki-numbers-nail.md:2]: Why a minor for auth? The API doesn't change. I don't think we bump automatically based on dependencies' bump level. I'm not totally sure if changesets does it but I think it also doesn't.

Comment (DellaBitta) [reply .changeset/khaki-numbers-nail.md]: Ok, switched Auth to a patch bump instead of a minor bump.

File: packages/auth/src/api/index.ts [hunk 2]
```diff
@@ -148,14 +148,19 @@ export async function _performApiRequest<T, V>(
       headers[HttpHeader.X_FIREBASE_LOCALE] = auth.languageCode;
     }
 
+    const fetchArgs: RequestInit = {
+      method,
+      headers,
+      ...body
+    };
+
+    if (!isCloudflareWorker()) {
```
Comment (Xiaoshouzi-gh) [inline packages/auth/src/api/index.ts:157]: can we add a comment here explaining this? Maybe we should link this github issue here https://github.com/cloudflare/next-on-pages/issues/487 and mention "Security-conscious server-side frameworks tend to have built in mitigations for referrer problems"

**Review by Xiaoshouzi-gh — APPROVED** [review]

File: packages/auth/src/api/index.test.ts [hunk 3]
```diff
@@ -308,6 +309,66 @@ describe('api/_performApiRequest', () => {
...
+      sinon.stub(utils, 'isCloudflareWorker').returns(true);
+      /* eslint-disable  no-var */
+      var referrerPolicySet: boolean = false;
+      /* eslint-enable  no-var */
+      mockFetch.setUpWithOverride(
+        (input: RequestInfo | URL, request?: RequestInit) => {
+          if (request !== undefined && request.referrerPolicy !== undefined) {
+            referrerPolicySet = true;
+          }
+          return new Promise<never>((_, reject) =>
```
Comment (Xiaoshouzi-gh) [inline packages/auth/src/api/index.test.ts:352]: optional: possible to resolve the promise instead of reject?

**Review by Xiaoshouzi-gh — APPROVED** [review]

**Review by hsubox76 — APPROVED** [review]

File: packages/auth/src/api/index.test.ts [hunk 4]
```diff
@@ -308,6 +309,66 @@ describe('api/_performApiRequest', () => {
...
+        FirebaseError,
+        'auth/network-request-failed'
+      );
+      expect(referrerPolicySet).to.be.true;
+    });
+
+    it('should not have referrerPolicy set on Cloudflare workers', async () => {
+      sinon.stub(utils, 'isCloudflareWorker').returns(true);
+      /* eslint-disable  no-var */
+      var referrerPolicySet: boolean = false;
```
Comment (hsubox76) [inline packages/auth/src/api/index.test.ts:345]: Why var?

File: packages/auth/src/api/index.test.ts (hunk 3 above)
Comment (hsubox76) [reply packages/auth/src/api/index.test.ts:348]: Also curious why these tests have a rejected promise.

Comment (DellaBitta) [reply packages/auth/src/api/index.test.ts]: Done! Updated tests so that these mocks resolve.

**Pull request merged**